go/chat/helper.go

```diff
@@ -18,168 +18,71 @@ import (
+	}
 }
 
-type sendHelper struct {
+type SendHelper struct {
 	utils.DebugLabeler
-
-	name        string
-	membersType chat1.ConversationMembersType
-	ri          chat1.RemoteInterface
-	ident       keybase1.TLFIdentifyBehavior
-
-	canonicalName string
-	topicName     string
-	tlfID         chat1.TLFID
-	convID        chat1.ConversationID
-	triple        chat1.ConversationIDTriple
-
 	globals.Contextified
-}
 
-func newSendHelper(g *globals.Context, name string, topicName string,
-	membersType chat1.ConversationMembersType, ident keybase1.TLFIdentifyBehavior, ri chat1.RemoteInterface) *sendHelper {
-	return &sendHelper{
-		Contextified: globals.NewContextified(g),
-		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "sendHelper", false),
-		name:         name,
-		topicName:    topicName,
-		membersType:  membersType,
-		ri:           ri,
-		ident:        ident,
-	}
+	ChatServer       *Server
+	ConversationID   chat1.ConversationID
+	TlfName          string
+	Triple           chat1.ConversationIDTriple
+	IdentifyBehavior keybase1.TLFIdentifyBehavior
 }
 
-func (s *sendHelper) SendText(ctx context.Context, text string) error {
-	body := chat1.NewMessageBodyWithText(chat1.MessageText{Body: text})
-	return s.SendBody(ctx, body, chat1.MessageType_TEXT)
+func NewSendHelperFromInfo(server *Server, tlfName string, conversationID chat1.ConversationID, triple chat1.ConversationIDTriple, identifyBehavior keybase1.TLFIdentifyBehavior) (*SendHelper, error) {
+	return &SendHelper{
+		Contextified:     globals.NewContextified(server.G()),
+		DebugLabeler:     utils.NewDebugLabeler(server.G().GetLog(), "sendHelper", false),
+		ChatServer:       server,
+		ConversationID:   conversationID,
+		TlfName:          tlfName,
+		Triple:           triple,
+		IdentifyBehavior: identifyBehavior,
+	}, nil
 }
 
-func (s *sendHelper) SendBody(ctx context.Context, body chat1.MessageBody, mtype chat1.MessageType) error {
-	ctx = Context(ctx, s.G(), s.ident, nil, NewIdentifyNotifier(s.G()))
-	if err := s.nameInfo(ctx); err != nil {
-		return err
-	}
-
-	if err := s.conversation(ctx); err != nil {
-		return err
-	}
-
-	return s.deliver(ctx, body, mtype)
-}
-
-func (s *sendHelper) nameInfo(ctx context.Context) error {
-	nameInfo, err := CtxKeyFinder(ctx, s.G()).Find(ctx, s.name, s.membersType, false)
-	if err != nil {
-		return err
-	}
-	s.tlfID = nameInfo.ID
-	s.canonicalName = nameInfo.CanonicalName
-
-	return nil
-}
-
-func (s *sendHelper) conversation(ctx context.Context) error {
-	kuid, err := CurrentUID(s.G())
-	if err != nil {
-		return err
-	}
-	uid := gregor1.UID(kuid.ToBytes())
-
-	oneChatPerTLF := true
-	convs, _, err := FindConversations(ctx, s.G(), s.DebugLabeler, s.remoteInterface, uid, s.canonicalName,
-		chat1.TopicType_CHAT, s.membersType, chat1.TLFVisibility_PRIVATE, s.topicName, &oneChatPerTLF)
+func NewSendHelper(ctx context.Context, server *Server, nclArg chat1.NewConversationLocalArg) (*SendHelper, error) {
```

Comment:
rationale for taking in `NewConversationLocalArg` is that it's exactly the information we need, so there's no point in exploding it.